src/main/java/dan200/computercraft/client/gui/FixedWidthFontRenderer.java

```diff
@@ -5,195 +5,318 @@
  */
 package dan200.computercraft.client.gui;
 
+import dan200.computercraft.client.FrameInfo;
+import dan200.computercraft.core.terminal.Terminal;
 import dan200.computercraft.core.terminal.TextBuffer;
+import dan200.computercraft.shared.util.Colour;
 import dan200.computercraft.shared.util.Palette;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.BufferBuilder;
 import net.minecraft.client.renderer.GlStateManager;
 import net.minecraft.client.renderer.Tessellator;
-import net.minecraft.client.renderer.texture.TextureManager;
 import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.client.renderer.vertex.VertexFormat;
 import net.minecraft.util.ResourceLocation;
 import org.lwjgl.opengl.GL11;
 
-import java.util.Arrays;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 
 public final class FixedWidthFontRenderer
 {
     private static final ResourceLocation FONT = new ResourceLocation( "computercraft", "textures/gui/term_font.png" );
-    public static final ResourceLocation BACKGROUND = new ResourceLocation( "computercraft", "textures/gui/term_background.png" );
 
-    public static final int FONT_HEIGHT = 9;
-    public static final int FONT_WIDTH = 6;
-
-    private static FixedWidthFontRenderer instance;
+    /**
+     * Like {@link DefaultVertexFormats#POSITION_TEX_COLOR}, but flipped. This is backported from 1.15, hence the
+     * custom format.
+     */
+    public static final VertexFormat POSITION_COLOR_TEX = new VertexFormat();
 
-    public static FixedWidthFontRenderer instance()
+    static
     {
-        if( instance != null ) return instance;
-        return instance = new FixedWidthFontRenderer();
+        POSITION_COLOR_TEX.addElement( DefaultVertexFormats.POSITION_3F );
+        POSITION_COLOR_TEX.addElement( DefaultVertexFormats.COLOR_4UB );
```

Comment:
Do we need the 4th colour component?

Reply:
No, but `BufferBuilder` does! I want to stick with using this, as it's kinda unavoidable on 1.15 anyway, so might as well be consistent.